src/python/pants/backend/python/interpreter_cache.py

```diff
@@ -88,13 +89,29 @@ def select_interpreter_for_targets(self, targets):
 
     if not allowed_interpreters:
       # Create a helpful error message.
+      all_interpreter_version_strings = sorted({
+        interpreter.version_string
+        # NB: self.setup() requires filters to be passed, or else it will use the global interpreter
+        # constraints. We allow any interpreter other than CPython 3.0-3.3, which is known to choke
+        # with Pants.
+        for interpreter in self.setup(filters=("CPython<3", "CPython>=3.3", "PyPy"))
```

Comment:
Reviewers, am I missing any other Python versions that Pants supports? IronPy or anything maybe?